BrainPortal/lib/cbrain_task_generators/templates/task_params.html.erb.erb

```diff
@@ -71,6 +71,49 @@
     single_file = <%= single_file ? "\"#{single_file['id']}\"" : "nil" %>
 %%>
 
+
+<%-
+  #
+  # Make maps for which parameters disable/require each other
+  #
+  # Generates a map from id to [ids affected by the key id] for a given action name
+  makeActorActionMap = lambda do |action|
+    params.inject({}){ |hmap,i| (res=i[action]) ? hmap.merge(i["id"]=>res) : hmap }
+  end
+  #
+  # Generates a map from id to [ids affecting the key id] for a given action name
+  makeReceiverActionMap = lambda do |action|
+    doesMap = makeActorActionMap.(action)
+    doneMap = doesMap.inject({}) do |map,(key,val)| # invert the relation above (active->passive)
+      for v in val do
+        if v.is_a?(Array)
+          (disabled_by, keys) = v
+          keys.each do |k|
+            if map[k].nil?
+              map[k] = {key.to_s => [disabled_by]}
+            else
+              if map[k][key].nil? then map[k][key] = [disabled_by] else map[k][key] << disabled_by end
+            end
+          end
+        else
+          if map[v].nil? then map[v] = [key] else map[v] << key end
+        end
+      end
+      map
+    end
+  end
+
+  #
+  # The actual maps: idIn -> [ids disabled/required by idIn]
+  disabledBy_map   = makeReceiverActionMap.("disables-inputs")
```

Comment:
Here you are computing maps that are never used (disabledBy_map and requires_map)